client/app/dbaas/logs/detail/streams/logs-streams.service.js

```diff
@@ -0,0 +1,387 @@
+    }
+
+    getSubscribedOptions (serviceName) {
+        return this.LogsOptionsService.getStreamSubscribedOptions(serviceName);
+    }
+
+    /**
+     * creates new stream with default values
+     *
+     * @returns stream object with default values
+     * @memberof LogsStreamsService
+     */
+    getNewStream () {
+        return {
+            data: {
+                coldStorageCompression: this.compressionAlgorithms[0].value,
+                coldStorageRetention: this.storageDurations[0].value,
+                coldStorageNotifyEnabled: true,
+                coldStorageEnabled: false,
+                webSocketEnabled: true
+            },
+            loading: false
+        };
+    }
+
+    /**
+     * extracts graylog URL from stream. Shows error message on UI if no graylog URL is found.
+     *
+     * @param {any} stream
+     * @returns {string} graylog url, if not found empty string
+     * @memberof LogsStreamsService
+     */
+    getStreamGraylogUrl (stream) {
+        const url = this.UrlHelper.findUrl(stream, this.LogStreamsConstants.GRAYLOG_WEBUI);
+        if (!url) {
+            this.CloudMessage.error(this.$translate.instant("logs_streams_get_graylog_url_error", { stream: stream.info.title }));
+        }
+        return url;
+    }
+
+    /**
+     * extracts and copies stream token to clipboard.
+     * Shows error message on UI if no no token found or browser does not support clipboard copy.
+     *
+     * @param {any} stream
+     * @memberof LogsStreamsService
+     */
+    copyStreamToken (stream) {
+        const token = this.getStreamToken(stream);
+        if (token) {
+            const error = this.ControllerHelper.copyToClipboard(token);
+            if (error) {
+                this.CloudMessage.error(this.$translate.instant("logs_streams_copy_token_error", {
+                    stream: stream.info.title,
+                    token_value: token
+                }));
+            } else {
+                this.CloudMessage.success(this.$translate.instant("logs_streams_copy_token_success"));
+            }
+        }
+    }
+
+    /**
+     * Extracts X-OVH-TOKEN token from given stream.
+     * Throws exception on UI if token was not found.
+     * @param {object} stream
+     * @return {string} stream token if found, empty string otherwise
+     */
+    getStreamToken (stream) {
+        const token = this.findStreamTokenValue(stream);
+        if (!token) {
+            this.CloudMessage.error(this.$translate.instant("logs_streams_find_token_error", { stream: stream.info.title }));
+        }
+        return token;
+    }
+
+    /**
+     * extracts X-OVH-TOKEN token from given stream
+     * @param {object} stream
+     * @return {string} stream token if found, empty string otherwise
+     */
+    findStreamTokenValue (stream) {
+        const ruleObj = _.find(stream.rules, rule => rule.field === this.LogStreamsConstants.X_OVH_TOKEN);
+        return _.get(ruleObj, "value", "");
```

Comment:
is it really important that the default value should be ""?  You could omit the third parameter otherwise

Reply:
_.get returns undefined if the default value is not explicitly given and if the property is not found in the object. But "" (default value) can be removed, since it is not really required. Hence it has been removed.